frameworks/helloworld/tests/test_region_awareness.py

```diff
@@ -0,0 +1,98 @@
+import json
+import logging
+
+import pytest
+import sdk_cmd
+import sdk_install
+import sdk_marathon
+import sdk_plan
+import sdk_utils
+from tests import config
+
+log = logging.getLogger(__name__)
+
+POD_NAMES = ['hello-0', 'world-0', 'world-1']
+
+
+@pytest.mark.dcos_min_version('1.11')
+@pytest.mark.region_awareness
+@sdk_utils.dcos_ee_only
+def test_nodes_deploy_to_local_region_by_default():
+    sdk_install.install(
+        config.PACKAGE_NAME,
+        config.SERVICE_NAME,
+        3,
+        additional_options={"service": {"scenario": "MULTI_REGION"}})
+
+    sdk_plan.wait_for_completed_deployment(config.SERVICE_NAME)
+
+    for pod_name in POD_NAMES:
+        info = sdk_cmd.service_request(
+            'GET', config.SERVICE_NAME, '/v1/pod/{}/info'.format(pod_name)
+        ).json()[0]['info']
+
+        assert (
+            [l['value'] for l in info['labels']['labels'] if l['key'] == 'offer_region'][0] == 'USA'
+        )
+    
+    sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+
+@pytest.mark.dcos_min_version('1.11')
+@pytest.mark.region_awareness
+@sdk_utils.dcos_ee_only
+def test_nodes_can_deploy_to_remote_region():
+    sdk_install.install(
+        config.PACKAGE_NAME,
+        config.SERVICE_NAME,
+        3,
+        additional_options={"service": {"scenario": "MULTI_REGION", "region": "Europe"}})
+
+    sdk_plan.wait_for_completed_deployment(config.SERVICE_NAME)
+
+    for pod_name in POD_NAMES:
+        info = sdk_cmd.service_request(
+            'GET', config.SERVICE_NAME, '/v1/pod/{}/info'.format(pod_name)
+        ).json()[0]['info']
+
+        assert (
+            [l['value'] for l in info['labels']['labels'] if l['key'] == 'offer_region'][0] ==
+            'Europe'
+        )
+
+    sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+
+@pytest.mark.dcos_min_version('1.11')
+@pytest.mark.region_awareness
+@sdk_utils.dcos_ee_only
+def test_region_config_update_does_not_succeed():
+    sdk_install.install(
+        config.PACKAGE_NAME,
+        config.SERVICE_NAME,
+        3,
+        additional_options={"service": {"scenario": "MULTI_REGION"}})
+
+    sdk_plan.wait_for_completed_deployment(config.SERVICE_NAME)
+    change_region_config('Europe')
+    plan = sdk_plan.get_deployment_plan(config.SERVICE_NAME)
+    assert plan.get('errors', [])
+
+    sdk_cmd.svc_cli(config.PACKAGE_NAME, config.SERVICE_NAME, 'pod replace hello-0')
+    sdk_plan.wait_for_completed_recovery(config.SERVICE_NAME)
+
+    info = sdk_cmd.service_request(
```

Comment:
Getting the pod information and asserting that the value is as desired could be perfect for a utility function.

Reply:
I think DRY is less important in test code -- better in my opinion to be able to read the test at a glance to see what's going on. Also since the JSON schema for pod info is both a broad and deep tree, I've always found it non-obvious which utility function is actually going to get me what I need.